transcom/mymove
# B 19086 int

## [B-19086](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19086)

## Summary

This is a story that was created due to finishing work for Sprint 1 early, made to reduce the number of incomplete values from the inherited SSW computations, since they could not be added while inserting the new PDF generation and template. The goal of this task is to help the project outline and prepare for a potential feature revamp with new requirements for the SSW to have all data fields complete. This ticket adds considerably more information into the document that reflects what should have been there when we inherited it. That said, there are still a few fields that this task does not cover, they are as follows: 

1. Disbursement: What exactly is desired for this field is still uncertain to the team.
2. Remaining PPM Incentive: What calculation is desired for this field is still uncertain to the team.
3. Enclosed Documentation Section: We have been instructed to ignore this section
4. PPPO/PPSO Signature: This signature is not currently being collected.
5. POV Shipment Authorized: This field is always going to be "No" until we are given information on how this decision should be made.

The added fields include: 

1. W2Address
2. Correct Signature date
3. The claimable expenses section 
4. The Agents section
6. Pro gear field
7. Claimable expenses gas - needs removed
8. Claimable expenses rental equipment - needs added
9. Member and GTCC paid SIT
10. Ensure font size is uniform across the entire document
11. Ensure Multiline is enabled in the document
12. SIT Shipment Number
13. Pickup Date
14. Obligations section - SIT Allowances tracked in Days

Is there anything you would like reviewers to give additional scrutiny?
Please ensure that when testing these fields, you take all required steps to fill them. This includes submitting moving expenses, signing off on the PPM Shipment (for the signature date), etc. I apologize that the testing process is somewhat tedious.

The agility ticket explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Log in as a new customer
2. Create a move and PPM Shipment, fill out all data fields possible, including requesting an advance
3. When you can go no farther in the process, log in as a services counselor for the same GLOC and approve the orders, advance, and ppm.
4. Log back in as the customer, continue on the move workflow. This is where you will give your W2 address, submit at least 1 expense report for each type, enter SIT data, etc.
5. Once complete, save and sign that all information is correct.
6. Then, access the AOA Packet (the SSW is generated as the first two pages of the AOA)
7. If the UI element isn't working in integration, use dev tools to find your ppm shipment id, and access the following endpoint: http://milmovelocal:3000/internal/ppm-shipments/eb0fdbaf-7f92-470d-a498-ed1e32ec8d36/aoa-packet with the correct ppm shipment id. 
8. There, you should see all updated fields and information filling correctly.
9. Ensure all fields are the same font size, and support multiline text (multiline not required for payment request section).
10. Run server tests, ensure passing.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
Examples of filled PDF: 
<img width="559" alt="image" src="https://github.com/transcom/mymove/assets/148906585/0ac85f17-2844-4d1c-8963-dfcf4699b3bd">
<img width="559" alt="image" src="https://github.com/transcom/mymove/assets/148906585/dfd7f699-500a-4d8f-80e4-bebc484be72c">
